src/views/AutogenView.vue

```diff
@@ -740,7 +740,8 @@ export default {
       this.currentAction.paramFields.map(field => {
         let fieldValue = null
         let fieldName = null
-        if (field.type === 'list' || field.name === 'account' || (this.currentAction.mapping && field.name in this.currentAction.mapping)) {
+        if (field.type === 'list' || field.name === 'account' ||
+          (this.currentAction.mapping && field.name in this.currentAction.mapping && !(field.name in this.resource))) {
```

Comment:
actually a field name could have a custom mapping and be in a resource (there are a few example), so this may break such cases; can you fix it by rather adding a mapping to the edit host action @shwstppr but retain the `this.currentAction.mapping && field.name in this.currentAction.mapping` check

Reply:
@rhtyd we already have a mapping for `oscategoryid` in `hosts.js` which is why we are seeing this problem.
https://github.com/apache/cloudstack-primate/blob/master/src/config/section/infra/hosts.js#L63-L67
While prefilling current values of params in edit forms, code is replacing `oscategoryid` to `oscategoryname` and using the value for `oscategoryname` from the resource.
IMO, it should not do this replacement just because we have a mapping for a param but since I'm not sure if removing it completely will break some case I just added an additional condition to stop this replacement when the original param (`oscategoryid` in `listHost` response) is available in the resource object.